modules/collabora_online_group/tests/src/Functional/ViewTest.php

```diff
@@ -0,0 +1,143 @@
+  use GroupTestTrait;
+  use MediaTypeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'collabora_online_group',
+    'views',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'stark';
+
+  /**
+   * Test the links for Collabora operations in the view.
+   */
+  public function testViewLinks(): void {
+    // Add configuration needed for testing.
+    $group_type = $this->createGroupType(['id' => 'group_type_1']);
+    $media_type = $this->createMediaType('file', ['id' => 'document']);
+    $this->createGroupRole([
+      'group_type' => 'group_type_1',
+      'scope' => PermissionScopeInterface::INSIDER_ID,
+      'global_role' => RoleInterface::AUTHENTICATED_ID,
+      'permissions' => [
+        'view group',
+        'access group_media overview',
+        'view group_media:document entity',
+        'edit any group_media:document in collabora',
+        'preview group_media:document in collabora',
+        ],
+    ]);
+    $this->createPluginRelation($group_type, 'group_media:document', [
+      'group_cardinality' => 0,
+      'entity_cardinality' => 1,
+      'use_creation_wizard' => FALSE,
+    ]);
+
+    // Create content.
+    $group = $this->createGroup(['type' => 'group_type_1']);
+    for ($i = 1;$i < 4;$i++) {
+      $media = $this->createMediaEntity('document', [
+        'id' => 'media_' . $i,
+        'name' => 'Media ' . $i,
+      ]);
+      $group->addRelationship($media, 'group_media:document');
+    }
+    $user = $this->createUser([
+      'view the administration theme',
+      'access administration pages',
+      'access group overview',
+    ]);
+    $group->addMember($user);
+
+    // Go to the page and check the links added to the view.
+    $this->drupalLogin($user);
+    $this->drupalGet("group/{$group->id()}/media");
+    $assert_session = $this->assertSession();
+
+    // Check table header.
+    $table = $assert_session->elementExists('css', 'table');
+    $table_header = $assert_session->elementExists('css', 'thead', $table);
+    $rows = $table_header->findAll('css', 'tr');
+    $cols = $rows[0]->findAll('css', 'th');
+    $this->assertEquals('Media name', $cols[0]->getText());
+    $this->assertEquals('Bundle', $cols[1]->getText());
+    $this->assertEquals('Status', $cols[2]->getText());
+    $this->assertEquals('Publisher', $cols[3]->getText());
+    // Support for different vesrions of groupmedia.
+    $this->assertEquals($this->isGroupRelationShipType() ? 'Operations' : 'Dropbutton', $cols[4]->getText());
+
+    // Check that rows contain new links for operations in Collabora.
+    $table_body = $assert_session->elementExists('css', 'tbody', $table);
+    $rows = $table_body->findAll('css', 'tr');
+    $i = 0;
+    foreach(Media::loadMultiple() as $media) {
+      $cols = $rows[$i]->findAll('css', 'td');
+      $this->assertEquals($media->getName(), $cols[0]->getText());
+      $this->assertEquals($media_type->label(), $cols[1]->getText());
```

Comment:
I would hard-code the media title and media type label.

Reply:
Why?

Reply:
Generally, the more we use hard-coded explicit values, the more predictable the test looks and behaves. It means less surprises, and less things the developer needs to know or think about when reading the test.

E.g. if we call `$media->getName()`, one could wonder if in some silly scenario the result might be NULL or a different value, because something went wrong, or we are using the wrong media. So we could have a passing test where both values are NULL or '', when in fact we want the test to fail.

This is why, when we should assert with explicit literal values, when there is no good reason to make it dynamic.

Reply:
Actually I set values specific values for the medias 😅